### PR TITLE
✨(courses) display related programs on course detail page

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

Courses can be linked on a program detail page. In this case, it is interesting to link the related programs on the page. 

## Proposal

I adapted what had been done to list related courses on an organization detail page.

closes https://github.com/openfun/richie/issues/1237

### CODE REVIEW

File: src/richie/apps/courses/models/course.py [hunk 1]
```diff
@@ -473,6 +474,40 @@ def get_course_runs_dict(self):
...
+        bfs = (
+            "extended_object__placeholders__cmsplugin__courses_coursepluginmodel__page"
+        )
+        filter_dict = {
+            "extended_object__publisher_is_draft": is_draft,
+            "extended_object__placeholders__cmsplugin__language": language,
+            bfs: course.extended_object,
+        }
+
+        program_model = apps.get_model(app_label="courses", model_name="program")
```
Comment (jbpenrath) [inline src/richie/apps/courses/models/course.py:495]: Just for my understanding: Why not just import directly the Program model from `.program import Program` as it has been done for category, organization etc. ?

File: src/richie/apps/courses/templates/courses/cms/course_detail.html [hunk 2]
```diff
@@ -329,6 +329,37 @@ <h3 class="course-detail__title">
...
+                        <div class="course-detail__row">
+                            <section class="course-glimpse-list">
+                                <h2 class="course-detail__title">
+                                    {% blocktrans count counter=programs|length %}
+                                        This course is part of a program
+                                    {% plural %}
+                                        This course is part of programs
+                                    {% endblocktrans %}
+                                </h2>
+                                <div class="section__items section__items--programs">
```
Comment (jbpenrath) [inline src/richie/apps/courses/templates/courses/cms/course_detail.html:346]: When there is only one program in the `section__items`, the layout is broken. I suggested you a fix 235cb6da427bdcb246f93ab23371feee10aa6520

Capture d’écran 2021-01-20 à 09 36 42
Capture d’écran 2021-01-20 à 09 45 51

File: tests/apps/courses/test_models_course.py [hunk 3]
```diff
@@ -341,11 +336,14 @@ def test_models_course_get_organizations(self):
         # course in the following, the third category will not be linked so we can check that
         # only the organizations linked to the course are retrieved (its name starts with `_`
         # because it is not used and only here for unpacking purposes)
-        *draft_organizations, _other_draft = OrganizationFactory.create_batch(3)
-        *published_organizations, _other_public = OrganizationFactory.create_batch(
-            3, should_publish=True
-        )
-        course = CourseFactory(
+        *draft_organizations, _other_draft = factories.OrganizationFactory.create_batch(
```
Comment (jbpenrath) [inline tests/apps/courses/test_models_course.py:339]: Sometimes lint rules don't do what they are supposed to do (in this case improve readability).

**Review by jbpenrath — COMMENTED** [review]
I suggested you a layout fix

File: src/richie/apps/courses/models/course.py (hunk 1 above)
Comment (sampaccoud) [reply src/richie/apps/courses/models/course.py]: I admit I copied it from somewhere else in the code. It is probably a relic of a generic code where the model name was passed as argument. I removed it to make it simpler.

File: tests/apps/courses/test_models_course.py (hunk 3 above)
Comment (sampaccoud) [reply tests/apps/courses/test_models_course.py:339]: :man_shrugging:

File: CHANGELOG.md [hunk 4]
```diff
@@ -20,6 +21,7 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ### Fixed
 
+- Deleted template `course_run_detail.html` was still referenced in settings
```
Comment (jbpenrath) [inline CHANGELOG.md:24]:
```suggestion
- Delete template `course_run_detail.html` was still referenced in settings
```

**Review by jbpenrath — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 4 above)
Comment (sampaccoud) [reply CHANGELOG.md]: :pray:

**Pull request merged**